homeassistant/components/cast/media_player.py

```diff
@@ -616,9 +611,10 @@ def audio_content_filter(item):
 
             content_filter = audio_content_filter
 
-        if media_content_id is None:
+        if media_content_id is None or media_content_type is None:
             return await self._async_root_payload(content_filter)
 
+        platform: CastProtocol
         for platform in self.hass.data[CAST_DOMAIN]["cast_platform"].values():
```

Comment:
```suggestion
        if media_content_id is None:
            return await self._async_root_payload(content_filter)

        platform: CastProtocol
        assert media_content_type is not None
        for platform in self.hass.data[CAST_DOMAIN]["cast_platform"].values():
```
`media_content_id` and `media_content_type` are inclusive in the same group. I would suggest to use an assert in these cases.
https://github.com/home-assistant/core/blob/f14bb8195ff2c5118b58b5959ab22a73a884c630/homeassistant/components/media_player/__init__.py#L1171-L1180

Reply:
There was a previous discussion where it was recommended to avoid `assert`/`cast`/`if TYPE_CHECKING` unless absolutely necessary.
In this case, I think my solution is better than using `assert`.